src/main/scala/cromwell/engine/workflow/CallMetadataBuilder.scala

```diff
@@ -0,0 +1,190 @@
+package cromwell.engine.workflow
+
+import cromwell.binding._
+import cromwell.engine.backend.{CallMetadata, StdoutStderr}
+import cromwell.engine.db.ExecutionDatabaseKey
+import cromwell.engine.db.slick._
+import cromwell.engine.ExecutionIndex._
+
+import scala.language.postfixOps
+
+/**
+ * Builds call metadata suitable for return as part of a workflow metadata request.
+ */
+object CallMetadataBuilder {
+
+  // Collects up all the data for a single CallMetadata in the easiest-to-assemble format, but
+  // not quite the format required for the metadata REST endpoint.
+  private case class AssembledCallMetadata(key: ExecutionDatabaseKey,
+                                           execution: Execution,
+                                           inputs: Traversable[Symbol] = Seq.empty,
+                                           outputs: Option[Traversable[Symbol]] = None,
+                                           streams: Option[StdoutStderr] = None,
+                                           backend: Option[String] = None,
+                                           jobId: Option[String] = None,
+                                           backendStatus: Option[String] = None)
+
+  // Types used in interim steps of the construction of the call metadata.
+  // Map from an `ExecutionDatabaseKey` to the interim `AssembledCallMetadata` format.
+  private type ExecutionMap = Map[ExecutionDatabaseKey, AssembledCallMetadata]
```

Comment:
Given our contemporary view on type aliases, do you think these are providing positive value? And really what I'm asking is if you think the benefit they provide is worth the hassle. 

I'm leaning towards "no" but could be easily convinced otherwise. They're not used in many places, are only used here (since they're private) and even add a few bits of less-than-ideal syntax (e.g. line 160). Since they don't provide any type safety I'm not sure the name abstraction is doing much. 

OTOH all of the non-committal phrases above are because I'm not sure I buy what I'm saying either :)